### PR TITLE
Scc 4233/add username to account header

### PR DESCRIPTION
**Pull request by charmingduchess**
## Ticket:

- JIRA ticket [SCC-4233](https://newyorkpubliclibrary.atlassian.net/browse/SCC-4233)

## This PR does the following:

- Fetches username from sierra
- displays username

## How has this been tested?

Updated unit tests and verified that usernames display. I don't have a patron account on hand with no username, so if anyone has one, that would be good to check!

## Accessibility concerns or updates

-

### Checklist:

- [x] I updated the CHANGELOG with the appropriate information and JIRA ticket number (if applicable).
- [ ] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

[SCC-4233]: https://newyorkpubliclibrary.atlassian.net/browse/SCC-4233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: __test__/fixtures/processedMyAccountData.ts [hunk 1]
```diff
@@ -317,6 +317,7 @@ export const filteredPickupLocations = [
 
 export const processedPatron = {
   notificationPreference: "z",
+  username: "spaghettimaster",
```
Comment (dgcohen) [inline __test__/fixtures/processedMyAccountData.ts:320]: nit: change to "stregadisciple"

**Review by dgcohen — APPROVED** [review]
Looks good and working locally. Just one minor comment.

**Comment by dgcohen** [comment]
Also out of curiosity, did we ever consider if patron has enough data and functionality to warrant its own class?

**Comment by charmingduchess** [comment]
@dgcohen the data is pretty static so i don't think it needs to be a class.

**Comment by dgcohen** [comment]
> @dgcohen the data is pretty static so i don't think it needs to be a class.

Yeah that makes sense, though you can argue the same for some of the other classes we have. It's definitely out of scope of this PR but it might be a good idea for us to start thinking about criteria for what goes in a model vs a utils file.

**Pull request merged**